presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -1629,6 +1640,61 @@ private void doTestPathHiddenColumn(Session session, HiveStorageFormat storageFo
     }
 
     @Test
+    public void testBucketNumberHiddenColumn()
+            throws Exception
+    {
+        @Language("SQL") String createTable = "CREATE TABLE test_bucket_number " +
+                "WITH (" +
+                "bucketed_by = ARRAY['col0']," +
+                "bucket_count = 2" +
+                ") AS " +
+                "SELECT * FROM (VALUES " +
+                "(0, 11), (1, 12), (2, 13), " +
+                "(3, 14), (4, 15), (5, 16), " +
+                "(6, 17), (7, 18), (8, 19)" +
+                " ) t (col0, col1) ";
+        assertUpdate(createTable, 9);
+        assertTrue(queryRunner.tableExists(getSession(), "test_bucket_number"));
+
+        TableMetadata tableMetadata = getTableMetadata(catalog, TPCH_SCHEMA, "test_bucket_number");
+        assertEquals(tableMetadata.getMetadata().getProperties().get(BUCKETED_BY_PROPERTY), ImmutableList.of("col0"));
+        assertEquals(tableMetadata.getMetadata().getProperties().get(BUCKET_COUNT_PROPERTY), 2);
+
+        List<String> columnNames = ImmutableList.of("col0", "col1", PATH_COLUMN_NAME, BUCKET_NUMBER_COLUMN_NAME);
+        List<ColumnMetadata> columnMetadatas = tableMetadata.getColumns();
+        assertEquals(columnMetadatas.size(), columnNames.size());
+        for (int i = 0; i < columnMetadatas.size(); i++) {
+            ColumnMetadata columnMetadata = columnMetadatas.get(i);
+            assertEquals(columnMetadata.getName(), columnNames.get(i));
+            if (columnMetadata.getName().equals(BUCKET_NUMBER_COLUMN_NAME)) {
+                // $bucket_number should be hidden column
+                assertTrue(columnMetadata.isHidden());
+            }
+        }
+        assertEquals(getBucketCount("test_bucket_number"), 2);
+
+        MaterializedResult results = computeActual(format("SELECT *, \"%s\" FROM test_bucket_number WHERE \"%s\"=1",
+                BUCKET_NUMBER_COLUMN_NAME,
```

Comment:
Or just inline the name. Once this goes in, we can't change it without breaking everyone. I think it's more clear to inline it.